plugins/utils/arrow.nix

```diff
@@ -0,0 +1,226 @@
+      '';
+
+      clear_all_items = helpers.defaultNullOpts.mkStr "C" ''
+        Mapping to clear all bookmarks.
+      '';
+
+      toggle = helpers.defaultNullOpts.mkStr "s" ''
+        Mapping to save if separate_save_and_remove is true.
+      '';
+
+      open_vertical = helpers.defaultNullOpts.mkStr "v" ''
+        Mapping to open bookmarks in vertical split.
+      '';
+
+      open_horizontal = helpers.defaultNullOpts.mkStr "-" ''
+        Mapping to open bookmarks in horizontal split.
+      '';
+
+      quit = helpers.defaultNullOpts.mkStr "q" ''
+        Mapping to quit arrow.
+      '';
+
+      remove = helpers.defaultNullOpts.mkStr "x" ''
+        Mapping to remove bookmarks. Only used if separate_save_and_remove is true.
+      '';
+
+      next_item = helpers.defaultNullOpts.mkStr "]" ''
+        Mapping to go to next bookmark.
+      '';
+
+      prev_item = helpers.defaultNullOpts.mkStr "[" ''
+        Mapping to go to previous bookmark.
+      '';
+    };
+
+    custom_actions = {
+      open =
+        helpers.defaultNullOpts.mkLuaFn
+          ''
+            function(target_file_name, current_file_name) end
+          ''
+          "target_file_name = file selected to be open, current_file_name = filename from where this was called";
+
+      split_vertical = helpers.defaultNullOpts.mkLuaFn ''
+        function(target_file_name, current_file_name) end
+      '' "";
+
+      split_horizontal = helpers.defaultNullOpts.mkLuaFn ''
+        function(target_file_name, current_file_name) end
+      '' "";
+    };
+
+    window = {
+      width = helpers.defaultNullOpts.mkStr "auto" ''
+        Width of the window.
+      '';
+
+      height = helpers.defaultNullOpts.mkStr "auto" ''
+        Height of the window.
+      '';
+
+      row = helpers.defaultNullOpts.mkStr "auto" ''
+        Row position in units of "screen cell height".
+      '';
+
+      col = helpers.defaultNullOpts.mkStr "auto" ''
+        Column position in units of "screen cell width".
+      '';
+
+      border = helpers.defaultNullOpts.mkBorder "double" ''
+        Border style of the window.
+      '' "";
+    };
+
+    per_buffer_config = {
+      lines = helpers.defaultNullOpts.mkInt 4 ''
+        Number of lines on preview.
+      '';
+
+      sort_automatically = helpers.defaultNullOpts.mkBool true ''
+        If true will sort buffer marks automatically.
+      '';
+
+      satellite = {
+        enable = helpers.defaultNullOpts.mkBool false ''
+          If true will display arrow index in scrollbar at every update.
+        '';
+
+        overlap = helpers.defaultNullOpts.mkBool false '''';
+
+        priority = helpers.defaultNullOpts.mkInt 1000 '''';
+      };
+
+      zindex = helpers.defaultNullOpts.mkInt 50 ''
+        Z index of the buffer.
+      '';
+
+      # currently not usable, for detail see https://github.com/otavioschwanck/arrow.nvim/pull/43#issue-2236320268
+      # treesitter_context = helpers.defaultNullOpts.mkNullable null '''';
```

Comment:
wasn't sure if i should include this or just remove it as the option doesn't work. I don't think the typing is correct either :thinking:

Reply:
I'd suggest not including it, you can mention in the PR/commit message that this option is _not_ handled because it is broken

Reply:
done